.circleci/config.yml

```diff
@@ -0,0 +1,75 @@
+version: 2
+jobs:
+  test:
+    working_directory: /go/src/github.com/Azure/acs-engine
+    docker:
+      - image: microsoft/acs-engine:latest
+    steps:
+      - setup_remote_docker:
+          reusable: true
+      - checkout
+      - run:
+          name: Build current image
+          command: |
+            TAG="0.1.${CIRCLE_BUILD_NUM}"
+            docker build -t microsoft/acs-engine:$TAG .
```

Comment:
I understand the desire to ensure that the docker image properly builds - which will check the Dockerfile itself functions.  What if we use just the acs-engine base plus `make bootstrap` to speed things up?

In this case you'd do all these same commands in the `test:` job without building the Docker image first because likely people are not going to be modifying the Dockerfile itself.  As long as `make bootstrap` successfully restores/updates the proper version of packages it should work the same on the newly created Docker image, yes?